src/lib/chips/chip-list.ts

```diff
@@ -85,14 +153,153 @@ export class MdChipList implements AfterContentInit, OnDestroy {
   /** The FocusKeyManager which handles focus. */
   _keyManager: FocusKeyManager<MdChip>;
 
-  /** The chip components contained within this chip list. */
-  chips: QueryList<MdChip>;
+  /** Function when touched */
+  _onTouched = () => {};
+
+  /** Function when changed */
+  _onChange: (value: any) => void = () => {};
 
-  constructor(protected _renderer: Renderer2, protected _elementRef: ElementRef,
-              @Optional() private _dir: Directionality) {
+  _selectionModel: SelectionModel<MdChip>;
+
+  /** Comparison function to specify which option is displayed. Defaults to object equality. */
+  private _compareWith = (o1: any, o2: any) => o1 === o2;
+
+  get selected(): MdChip[] {
+    return this._selectionModel.selected;
+  }
+
+  /**
+   * A function to compare the option values with the selected values. The first argument
+   * is a value from an option. The second is a value from the selection. A boolean
+   * should be returned.
+   */
+  @Input()
+  get compareWith() { return this._compareWith; }
+  set compareWith(fn: (o1: any, o2: any) => boolean) {
+    this._compareWith = fn;
+    if (this._selectionModel) {
+      // A different comparator means the selection could change.
+      this._initializeSelection();
+    }
+  }
```

Comment:
Could we share more code with the select here? Maybe a mixin that adds compareWith and default compareWith?